fix(flamegraph): correct diff formatting in flame graph diff tooltip

mrigueti opened the pull request:
fix: (flamegraph): correct diff formatting in flame graph diff tooltip

Large percentage values in the diff tooltip are now properly formatted with a 'K%' suffix (e.g., 8081.25% becomes 8.08K%).  
Also ensures thousand separators and decimal formatting match test expectations.

Related to #105391.

Commit:
fix(flamegraph): correct diff formatting in flame graph diff tooltip

Large percentage values in the diff tooltip are now properly formatted with a 'K%' suffix (e.g., 8081.25% becomes 8.08K%).
Also ensures thousand separators and decimal formatting match test expectations.

Related to #105391.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=105590) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=105590) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=105590) it.</sub>

ifrost commented on packages/grafana-flamegraph/src/FlameGraph/FlameGraphTooltip.tsx:
```diff
@@ -148,7 +158,7 @@ export const getDiffTooltipData = (
       baseline: percentageLeft + '%',
       comparison: percentageRight + '%',
-      diff: shortValFormat(diff).text + '%',
+      diff: formatDiff(diff),
```
I think we can use the existing formatter but we just need to add the suffix which is provided with the text value in a separate prop (https://github.com/grafana/grafana/blob/306ae8b4f531739f59a8465ba5095a10b55b21b7/packages/grafana-data/src/valueFormats/valueFormats.ts#L13). This way we reuse existing code and [get other suffixes out of the box](https://github.com/grafana/grafana/blob/032f465ac676853573c690b7136fdbb5bb09d275/packages/grafana-data/src/valueFormats/categories.ts#L44).

mrigueti replied:
I made the adjustments. Sorry if anything is out of order or wrong, I'm kind of new at this and I'm accepting feedback so I can improve.

ifrost replied:
No worries! I haven't had a chance to run it locally yet but left a few comments on what could make it work. Thanks again for having a look at this 🙌

Commit:
fix(flamegraph): format diff percentage in tooltip using shortValFormat

- Uses shortValFormat to format the diff value in the flamegraph tooltip.
- The '%' suffix is appended after formatting, ensuring consistency with other displayed numeric values.
- Adjustment made per code review suggestion, reusing the existing formatter and keeping the code clean and aligned with project standards.

ifrost commented on packages/grafana-flamegraph/src/FlameGraph/FlameGraphTooltip.tsx:
```diff
@@ -141,14 +141,16 @@ export const getDiffTooltipData = (
       baseline: percentageLeft + '%',
       comparison: percentageRight + '%',
-      diff: shortValFormat(diff).text + '%',
+      diff: formatted.text + '%',
```
The suffix is a separate property so we still need to add it (though we need to handle if it's not provided):

```suggestion
      diff: `${formatted.text}${formatted.suffix || ''}%`,
```
